components/lib/crash/src/main/java/mozilla/components/lib/crash/Crash.kt

```diff
@@ -78,17 +79,16 @@ sealed class Crash {
      * @property extrasPath Path to a file containing extra metadata about the crash. The file contains key-value pairs
      *                      in the form `Key=Value`. Be aware, it may contain sensitive data such as the URI that was
      *                      loaded at the time of the crash.
-     * @property isFatal Whether or not the crash was fatal or not: If true, the main application process was affected
-     *                   by the crash. If false, only an internal process used by Gecko has crashed and the application
-     *                   may be able to recover.
+     * @property processType The type of process the crash occurred in. Affects whether or not the crash is fatal
+     *                       or whether the application can recover from it.
      * @property breadcrumbs List of breadcrumbs to send with the crash report.
      */
     data class NativeCodeCrash(
         val timestamp: Long,
         val minidumpPath: String?,
         val minidumpSuccess: Boolean,
         val extrasPath: String?,
-        val isFatal: Boolean,
+        @ProcessType val processType: String?,
```

Comment:
Okay. Does it sound okay to have it as a function, calculated from `processType == MAIN`? (Can it even be a getter property so syntactically there's no change? Does the fact this is a data class prevent that?)

The `fatal` argument from the geckoview crash intent is deprecated and instead we use the process type. I agree `isFatal` is useful within android-components code, but it'd be nice to have it calculated from the process type to avoid any source of inconsistency.

Reply:
Yes, that's a good idea.  Setting `isFatal` with a function is a good solution.  This way, we don't have to break the API and also we can simplify a lot of checks to see if the crash is fatal.  Thanks